examples/leasing/node_release_script.py

```diff
@@ -0,0 +1,241 @@
+
+
+class HILClientFailure(Exception):
+    """Exception indicating that the HIL client failed"""
+
+
+def hil_client_connect(endpoint_ip, name, pw):
+    """Connect to the HIL server and return a HIL Client instance"""
+
+    hil_http_client = RequestsHTTPClient()
+    if not hil_http_client:
+        print('Unable to create HIL HTTP Client')
+        return None
+
+    hil_http_client.auth = (name, pw)
+
+    return Client(endpoint_ip, hil_http_client)
+
+
+def release_nodes(
+        statusfile, non_persistent_list,
+        threshold_time, hil_username, hil_password, hil_endpoint,
+        hil_client=None
+        ):
+
+    """Release nodes : Takes node list as arguments
+    and puts them back to free pool after given time"""
+
+    if not hil_client:
+        hil_client = hil_client_connect(
+                        hil_endpoint, hil_username, hil_password)
+
+    free_node_list = hil_client.node.list('free')
+    # Only these nodes should be updated in
+    # the file(either for project or for time)
+    nodes_to_update_infile = list(Set(
+        non_persistent_list)-Set(free_node_list))
+
+    for node in nodes_to_update_infile:
+        # get the information from status file for node.
+        node_info_from_file = get_project_and_time_for_node(statusfile, node)
+        project_in_file = node_info_from_file[0]
+        old_time = node_info_from_file[1]
+
+        # get the information from HIL for node
+        node_current_info = hil_client.node.show(node)
+        project_in_hil = node_current_info['project']
+        # Increase the time by one unit
+        new_time = int(node_info_from_file[1])+1
+        # See if the node is in the same project and if it is outside
+        # free pool for time less than threshold value then update only
+        # time in status file
+        if (project_in_file == project_in_hil and
+                new_time < threshold_time):
+            update_time_for_node(statusfile, node, old_time, new_time)
+        # See if the node is in the same project and if it is outside
+        # free pool for time greater than threshold value then release
+        # the node back to free pool
+        elif (project_in_file == project_in_hil and
+                new_time >= threshold_time):
+            # Detaching nodes from networks
+            detach_networks_from_node(hil_client, node)
+            # Releasing node from project.
+            release_from_project(hil_client, statusfile, node, project_in_hil)
+
+        # See if the node is not in free pool and check if the projects
+        # are not matching then we need to update project in status file
+        elif (project_in_file != project_in_hil and
+                project_in_hil is not None):
+            update_project_in_status_file(
+                statusfile, node, project_in_hil,
+                project_in_file)
+
+
+def get_project_and_time_for_node(statusfile, node):
+    """Reads current information from file for every node."""
```

Comment:
Kinda confused by the description and what the code seems to be doing. 

1. If this method is for returning info about one single node then:
* you should update the docstring to reflect that. 
* break the `for` loop once you find your match `node == node_status[0]`.

2. If this method is reading information for **every** node. Then you should just iterate over the file, and return a dictionary with `node_label`, `node_project`, and `node_duration`. passing `node` wouldn't be needed.